core_impl/src/main/java/io/opencensus/stats/MutableViewData.java

```diff
@@ -206,41 +191,148 @@ public MutableAggregation apply(StdDev arg) {
   @VisibleForTesting
   static AggregationData createAggregationData(MutableAggregation aggregation) {
     return aggregation.match(
-        new Function<MutableSum, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableSum arg) {
-            return SumData.create(arg.getSum());
-          }
-        },
-        new Function<MutableCount, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableCount arg) {
-            return CountData.create(arg.getCount());
-          }
-        },
-        new Function<MutableHistogram, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableHistogram arg) {
-            return HistogramData.create(arg.getBucketCounts());
-          }
-        },
-        new Function<MutableRange, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableRange arg) {
-            return RangeData.create(arg.getMin(), arg.getMax());
-          }
-        },
-        new Function<MutableMean, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableMean arg) {
-            return MeanData.create(arg.getMean());
-          }
-        },
-        new Function<MutableStdDev, AggregationData>() {
-          @Override
-          public AggregationData apply(MutableStdDev arg) {
-            return StdDevData.create(arg.getStdDev());
-          }
-        });
+        CreateSumData.INSTANCE,
+        CreateCountData.INSTANCE,
+        CreateHistogramData.INSTANCE,
+        CreateRangeData.INSTANCE,
+        CreateMeanData.INSTANCE,
+        CreateStdDevData.INSTANCE);
+  }
+  
+  // static inner Function classes
+  
+  private static final class CreateMutableSum implements Function<Sum, MutableAggregation> {
```

Comment:
It would be nice to be able to use the MutableAggregation subclass for the second type argument here.  I wonder if we should change the match method signature to accept more return types, like this:

```java
  public abstract <T> T match(
      Function<? super Sum, ? extends T> p0,
      Function<? super Count, ? extends T> p1,
      ...
);
```

I don't know whether that would cause any problems with type inference, though.  I'm not suggesting it for this PR.